### PR TITLE
work with ax3l: fail when fortran c interface could not be found

### PR DESCRIPTION
**Pull request by JBlaschke**
Following #1126, worked with @ax3l to make cmake give a informative error message and then fail (always a good thing to check return codes)

### CODE REVIEW

**Review by ax3l — APPROVED** [review]
I am @ax3l and I approve this message.

**Review by mic84 — APPROVED** [review]

File: Tools/CMake/AMReX_Defines.cmake [hunk 1]
```diff
@@ -93,6 +93,9 @@ function ( set_amrex_defines )
       # Fortran/C mangling scheme
       #
       include( FortranCInterface )
+      if(NOT FortranCInterface_GLOBAL_FOUND)
+          message(FATAL_ERROR "Failed to find the Fortan C Interface -- check the CMakeError.log")
```
Comment (ax3l) [inline Tools/CMake/AMReX_Defines.cmake:97]: Oh dang @JBlaschke: Fort**r**an

Comment (JBlaschke) [reply Tools/CMake/AMReX_Defines.cmake:97]: Eh @ax3l -- if they wanted people to spell the name right, they would have made it a single-letter

Comment (ax3l) [reply Tools/CMake/AMReX_Defines.cmake:97]: as long as it's all caps I am in.
F

Comment (JBlaschke) [reply Tools/CMake/AMReX_Defines.cmake:97]: As long as J will be "Julia" instead of "Java", then count me in!

**Comment by ax3l** [comment]
Founds something random about IBM XL today with respect to this. Just to document here:
> IBM xlf does not mangle Fortran symbols by default,use -qextnameto add trailing underscores.

https://www.olcf.ornl.gov/wp-content/uploads/2018/12/summit_workshop_20181203_programming_environment.pdf

**Pull request merged**